src/features/canvas/canvas-slice.ts

```
@@ -0,0 +1,288 @@
+                    return {
+                        config: removeNodeFromTree(state.config, id),
+                        positions: newPositions,
+                        hasUnsavedChanges: true
+                    }
+                }),
+
+            // Linking System
+            startLinking: (id, type) => set({ linkingNode: { id, type } }),
+            finishLinking: targetId => {
+                const state = get()
+                const { addToast } = useToast()
```

ESLint

[error] React Hook "useToast" is called in function "finishLinking" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".

(react-hooks/rules-of-hooks)